### PR TITLE
Add a timer-style variable

### PR DESCRIPTION
**Pull request by sjplimp**
**Summary**

Add timer-style variables to the variable class, for use in an input script.  They store the current elapsed CPU each time the
"variable name timer" command is processed in the script.  Thus using two timer-style variables to bracket an operation allows a print statement to output the difference of the 2 variable, to time the operation.  E.g.

variable start timer
other commands
variable stop timer
print "Elapsed time: $(v_stop-v_start:%.6f)"

**Related Issue(s)**

N/A

**Author(s)**

Steve

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: src/variable.cpp [hunk 1]
```diff
@@ -500,6 +500,31 @@ void Variable::set(int narg, char **arg)
...
+  // num = 1, for string representation of dvalue, set by retrieve()
+  // dvalue = numeric initialization via platform::cputime()
+
+  } else if (strcmp(arg[1],"timer") == 0) {
+    if (narg != 2) error->all(FLERR,"Illegal variable command");
+    int ivar = find(arg[0]);
+    if (ivar >= 0) {
+      if (style[ivar] != TIMER)
+        error->all(FLERR,"Cannot redefine variable as a different style");
+      dvalue[ivar] = platform::cputime();
```
Comment (akohlmey) [inline src/variable.cpp:515]: @sjplimp I am surprised you are using platform::cputime() here and not platform::walltime() or MPI_Wtime().

**Comment by sjplimp** [comment]
@akohlmey I thought cputime() was the equivalent of MPI_Wtime().  I would have used MPI_Wtime() except I don't see it anywhere in the code anymore.  What I want is something that returns a double in secs and the difference of 2 of them is the elapsed wall-clock time (to high precision).

**Comment by akohlmey** [comment]
> @akohlmey I thought cputime() was the equivalent of MPI_Wtime(). I would have used MPI_Wtime() except I don't see it anywhere in the code anymore. What I want is something that returns a double in secs and the difference of 2 of them is the elapsed wall-clock time (to high precision).

`MPI_Wtime()` has been superseded with `platform::walltime()`. This was an attempt to future-proof walltime measurements and avoid dependence on epoch. With storing the seconds in doubles the precision of the delta will significantly decline in the not so distant future due to the growing absolute value of the time since epoch in seconds as returned by MPI_Wtime(). platform::walltime() uses a high-precision timestamp when the executable was started as reference instead and thus there will be no loss of precision (unless the C++ library implementation of the clock functionality is bad).

Ok. I'll change it then.

**Comment by akohlmey** [comment]
FYI, https://docs.lammps.org/Developer_platform.html#time-functions

**Review by akohlmey — APPROVED** [review]

**Pull request merged**